### PR TITLE
Add Carbon Black Collector

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-6994

## Description
Add Carbon Black Collector.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by eepstain — APPROVED** [review]

**Review by eepstain — APPROVED** [review]

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                client=client,
+                last_run=last_run,
+                add_audit_logs=add_audit_logs,
+            )
+            demisto.debug(f'sending {len(events)} to xsiam')
+            if should_push_events:
+                send_events_to_xsiam(events=events, vendor=vendor, product=product)
+            demisto.debug(f'Handled {len(events)} total events')
+            demisto.setLastRun(new_last_run)
+
```
Comment (thefrieddan1) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector.py:258]: else : throw command not supported exception

**Review by thefrieddan1 — APPROVED** [review]
LGTM 2 minor comments

**Comment by DeanArbel** [comment]
@ShirleyDenkberg Can you please review the doc files?

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  section: Collect
+- display: Maximum number of alerts per fetch
+  additionalinfo: Default 100,000.
+  defaultvalue: '100000'
+  name: max_alerts
+  type: 0
+  required: false
+  section: Collect
+- display: Maximum number of audit logs per fetch
+  additionalinfo: Default 25,000.
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector.yml:52]:
```suggestion
  additionalinfo: Default 25000.
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+## Creating a API Key:
+1. Create *access level* entry with the following **READ** permissions:
+* org.alerts.notes
+* org.xdr.metadata
+* org.alerts
+* org.alerts.tags
+* org.audits
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md:7]:
```suggestion
   * org.alerts.notes
   * org.xdr.metadata
   * org.alerts
   * org.alerts.tags
   * org.audits
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+## Creating a API Key:
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md:1]:
```suggestion
## Create an API Key
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+1. Create *access level* entry with the following **READ** permissions:
+* org.alerts.notes
+* org.xdr.metadata
+* org.alerts
+* org.alerts.tags
+* org.audits
+
+2. Create the API key ([see documentation](https://developer.carbonblack.com/reference/carbon-black-cloud/authentication#authenticate-your-request)) with the following parameters:
+ * Access Level Type = Custom
+ * Custom Access Level = the name of the *access level* entry created in step 1.
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md:11]:
```suggestion
    * Access Level Type = Custom
    * Custom Access Level = the name of the *access level* entry created in step 1.
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+## Fetching Audit Logs Important Note:
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md:13]:
```suggestion
## Fetching Audit Logs Important Note
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+* The API for audit logs sends consumable audit log entries starting from 3 days before the creation of the API Key. If you wish to fetch the latest audit logs available, it is recommended to create a new API key for this instance, as the instance will create entries for the earliest available logs first.
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/CarbonBlackEndpointStandardEventCollector_description.md:14]:
```suggestion
* The API for audit logs sends consumable audit log entries starting from 3 days before the creation of the API Key. To fetch the latest audit logs available, it is recommended to create a new API key for this instance, as the instance will create entries for the earliest available logs first. 
```

File: Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+Endpoint Standard (formerly called Carbon Black Defense), a Next-Generation Anti-Virus + EDR. Collect Anti-Virus & EDR alerts and Audit Log Events.
+This integration was integrated and tested with version xx of CarbonBlackEndpointStandardEventCollector.
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/Integrations/CarbonBlackEndpointStandardEventCollector/README.md:2]: What is the version #?  (version xx)

File: Packs/CarbonBlackDefense/ReleaseNotes/3_1_0.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+#### Integrations
+
+##### New: Carbon Black Endpoint Standard Event Collector
+
+- New: Collect Anti-Virus & EDR alerts and Audit Log Events. (Available from Cortex XSIAM 2.1).
```
Comment (ShirleyDenkberg) [inline Packs/CarbonBlackDefense/ReleaseNotes/3_1_0.md:5]:
```suggestion
New: Collect Anti-Virus & EDR alerts and Audit Log Events. (Available from Cortex XSIAM 2.1).
```

**Comment by ShirleyDenkberg** [comment]
@thefrieddan1 @eepstain Doc review completed.

**Pull request merged**